cli/src/lib.rs

```diff
@@ -826,16 +844,32 @@ fn init(
     fs::write(".prettierignore", rust_template::prettier_ignore())?;
 
     // Build the program.
+    if force {
+        if solidity {
+            remove_dir_all(std::env::current_dir()?.join("solidity").join(&project_name))?;
+        } else {
+            remove_dir_all(std::env::current_dir()?.join("programs").join(&project_name))?;
+        }
+    }
+
     if solidity {
         solidity_template::create_program(&project_name)?;
     } else {
         rust_template::create_program(&project_name, template)?;
     }
 
     // Build the test suite.
-    fs::create_dir("tests")?;
+    if force {
+        fs::create_dir_all("tests")?;
+    } else {
+        fs::create_dir("tests")?;
+    }
```

Comment:
Does this change(and similar others) make any difference?

Reply:
Yes. `create_dir` will throw an error if the directory already exists but create_dir_all will not. Hence, in order to block execution if `--force` is not passed, we should use `create_dir` instead of `create_dir_all`. There are other ways to achieve the same outcome (error handling, directory validation, etc) but I believe this is a good usage of the two options.

<img width="854" alt="Screenshot 2023-11-10 at 22 16 00" src="https://github.com/coral-xyz/anchor/assets/97734034/20fc6d9b-639d-4a70-9874-f629fc05caed">

Reply:
Thanks, sorry I was looking at https://github.com/coral-xyz/anchor/pull/2698/commits/692a5933763d91bb57cb8c3daa5eae40803d4b6d#diff-c1f8f7498da827a634bddc8a7559198bc99b296e9d9e8b91a70b503662995b8cL846-R866 but commented on overall diff

```diff
- fs::create_dir_all("tests")?;
+ if force {
+     fs::create_dir_all("tests")?;
+ } else {
+     fs::create_dir("tests")?;
+ }
```

So my question was actually about this change — why not use `create_dir_all` here since all subfolders will be created for the first time for `init`(without `--force`) which should work for both cases?

> Hence, in order to block execution if `--force` is not passed, we should use `create_dir` instead of `create_dir_all`.

Isn't this check only necessary for the project root folder because if `--force` is not passed, all subfolders(`app`, `tests`, `migrations`) will be created for the first time anyway.

Reply:
> So my question was actually about this change — why not use create_dir_all here since all subfolders will be created for the first time for init(without --force) which should work for both cases?
> Isn't this check only necessary for the project root folder because if --force is not passed, all subfolders(app, tests, migrations) will be created for the first time anyway.

I kept `create_dir` for internal subfolders for consistency with the other flows but you're absolutely right that without `--force` it wouldn't execute either way so let's remove it. I'll open a PR in just a second.

And thank you so much for the thorough review, hopefully it will go smoother in future PR's as I get used to the project.

Reply:
Change done at https://github.com/coral-xyz/anchor/pull/2698/commits/488c012e9fe1f57fa064c6a7826844ff606e017c